src/features/messages/cards/TransactionCard.tsx

```diff
@@ -236,11 +242,64 @@ function CallDataModal({ debugResult }: { debugResult?: MessageDebugResult }) {
           </p>
           <LabelAndCodeBlock label="Recipient contract address:" value={contract} />
           <LabelAndCodeBlock label="Handle function input calldata:" value={handleCalldata} />
+          <button onClick={async()=>{
+            handleClick()
+            await simulateCall({contract,handleCalldata,chainId})
+          }}
+            disabled={disabled}
+            className='underline text-blue-400'
+          >
+            {buttonText}
+          </button>
+
         </div>
       </Modal>
     </>
   );
 }
+const simulateCall=async({contract,handleCalldata,chainId}:{contract:string,handleCalldata:string,chainId:ChainId})=>{
```

Comment:
Root-level functions should use `function` syntax please. (e.g. `async function simulateCall`)